### PR TITLE
Change "whitelist" to "allowlist"

### LINKED ISSUE #31064: [RFC] Remove blacklist and other oppressive terminology
I just stumbled opon the new [Bootstrap 5 alpha page](https://v5.getbootstrap.com/docs/5.0/getting-started/javascript/#sanitizer) and saw some words that could be improved to a more natural / tech specific wording, e.g. the `allowlist` and `blocklist` instead of `whitelist` and `blacklist`.

There are many projects that try to ban oppressive words from their code. This could be a great chance when releasing BS5

Refs: https://tools.ietf.org/id/draft-knodel-terminology-00.html, https://localheinz.com/blog/2020/06/12/black-lives-matter/

### PR DESCRIPTION
**Pull request by patrickhlauke**
Closes #31064

note that this leaves intact the use of `blacklist` for stylelint

### CODE REVIEW

**Review by Johann-S — APPROVED** [review]

**Review by XhmikosR — CHANGES_REQUESTED** [review]
Please don't merge I want to have a closer look.

**Comment by XhmikosR** [comment]
From a quick look, the change needs to be listed in the migration page in detail.

I will check later the branch.

**Comment by patrickhlauke** [comment]
ah, was thinking about migration guide, but then thought that it's not really a user/developer facing object (unless somebody was hooking directly into the function for some reason).

**Review by core23 — APPROVED** [review]

**Comment by XhmikosR** [comment]
> ah, was thinking about migration guide, but then thought that it's not really a user/developer facing object (unless somebody was hooking directly into the function for some reason).

Yeah, it's a breaking change so it needs to be documented. :)

**Comment by XhmikosR** [comment]
Pending the Migration addition, otherwise the changes look good.

**Comment by patrickhlauke** [comment]
added migration guide entry ... too detailed or ok? @XhmikosR

**Comment by XhmikosR** [comment]
I'd like @Johann-S opinion, not sure if all of the listed methods are public. We should only list public stuff.

**Comment by XhmikosR** [comment]
AFAICT only `allowList` is public.

File: site/content/docs/5.0/components/popovers.md [hunk 1]
```diff
@@ -142,15 +142,15 @@ To allow keyboard users to activate your popovers, you should only add them to H
...
 {{< /callout >}}
 
 ### Options
 
 Options can be passed via data attributes or JavaScript. For data attributes, append the option name to `data-`, as in `data-animation=""`.
 
 {{< callout warning >}}
-Note that for security reasons the `sanitize`, `sanitizeFn` and `whiteList` options cannot be supplied using data attributes.
+Note that for security reasons the `sanitize`, `sanitizeFn` and `allowList` options cannot be supplied using data attributes.
```
Comment (mdo) [inline site/content/docs/5.0/components/popovers.md:153]:
```suggestion
Note that for security reasons the `sanitize`, `sanitizeFn`, and `allowList` options cannot be supplied using data attributes.
```

File: site/content/docs/5.0/components/tooltips.md [hunk 2]
```diff
@@ -145,7 +145,7 @@ Elements with the `disabled` attribute aren't interactive, meaning users cannot
 Options can be passed via data attributes or JavaScript. For data attributes, append the option name to `data-`, as in `data-animation=""`.
 
 {{< callout warning >}}
-Note that for security reasons the `sanitize`, `sanitizeFn` and `whiteList` options cannot be supplied using data attributes.
+Note that for security reasons the `sanitize`, `sanitizeFn` and `allowList` options cannot be supplied using data attributes.
```
Comment (mdo) [inline site/content/docs/5.0/components/tooltips.md:148]: Should this be turned into a partial?

```suggestion
Note that for security reasons the `sanitize`, `sanitizeFn`, and `allowList` options cannot be supplied using data attributes.
```

Comment (patrickhlauke) [reply site/content/docs/5.0/components/tooltips.md]: not sure about partial ... it's (currently) only used twice ... would that be too much modularisation? no strong feelings about it either way though...let me know, happy to rip this out into partial

**Review by mdo — APPROVED** [review]
Thanks for this!

**Comment by patrickhlauke** [comment]
@Johann-S thoughts on the migration guide question? does it look good or does it need to be expanded?

**Comment by Johann-S** [comment]
LGTM on the migration guide, since it's a public option it's important to be added in our migration guide

**Review by XhmikosR — APPROVED** [review]

**Comment by datakid** [comment]
Note that stylelint removed `*-blacklist` https://github.com/stylelint/stylelint/releases/tag/13.7.0 - can this be revisited?

**Pull request merged**